### PR TITLE
Phase2-hgx316C Add a few utility methods in HGCalDDDConstants

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a few utility methods in HGCalDDDConstants

#### PR validation:

Use the runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/interface/HGCalDDDConstants.h [hunk 1]
```diff
@@ -86,6 +87,9 @@ class HGCalDDDConstants {
   int layerIndex(int lay, bool reco) const;
   unsigned int layers(bool reco) const;
   unsigned int layersInit(bool reco) const;
+  int layerType(int lay) const {
+    return ((hgpar_->layerType_.empty()) ? HGCalTypes::WaferCenter : hgpar_->layerType_[lay - hgpar_->firstLayer_]);
+  }
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/interface/HGCalDDDConstants.h:92]: CMS coding rule [4.3:](http://cms-sw.github.io/cms_coding_rules.html#4--technical-coding-rules-1)
Header ﬁles must not contain any implementation except for class templates and code to be inlined.

Add `inline` if you intend these methods to be inlined.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
+Upgrade

**Comment by bsunanda** [comment]
@srimanob @perrotta @qliphy Somehow Phat's signature is not recognized. Please merge this

**Comment by qliphy** [comment]
+1

**Pull request merged**